libs/client.go

```
@@ -68,6 +75,73 @@ func NewClient() *Client {
 	return &client
 }
 
+// SetProxy adds a proxy to the client
+func (c *Client) SetProxy(proxyList []string) {
```

The signature should return an error if no valid proxy was found